build-tools/packages/build-infrastructure/src/fluidRepo.ts

```diff
@@ -235,3 +237,52 @@ export function getAllDependenciesInRepo(
 		workspaces: [...workspaces],
 	};
 }
+
+/**
+ * Sets the dependency range for a set of packages given a group dependencies to update.
+ * This function will update the package.json files.
+ *
+ * Note that any loaded objects such as an IFluidRepo instance may need to be reloaded after calling this function.
+ *
+ * @param packagesToUpdate - An array of objects whose version should be updated.
+ * @param dependencies - An array of objects that the packagesToUpdate depend on.
+ * @param dependencyRange - The new version range to set for the packageToUpdate dependencies.
+ */
+export async function setDependencyRange(
+	packagesToUpdate: Iterable<IPackage>,
+	dependencies: Iterable<IPackage>,
+	dependencyRange: InterdependencyRange,
+): Promise<void> {
```

Comment:
```suggestion
export async function setDependencyRange<P extends IPackage>(
	packagesToUpdate: Iterable<P>,
	dependencies: Iterable<P>,
	dependencyRange: InterdependencyRange,
): Promise<void> {
```

I think making this generic will make it easier to use in different places, with different IPackage implementations, without losing the type information.